components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationContextCache.java

```diff
@@ -51,6 +53,9 @@ private AuthenticationContextCache() {
             isTemporarySessionDataPersistEnabled = Boolean.parseBoolean(
                     IdentityUtil.getProperty("JDBCPersistenceManager.SessionDataPersist.Temporary"));
         }
+        if (!FrameworkUtils.isSessionDataStorageOptimizationEnabled()) {
+            isSessionDataStorageOptimizationEnabled = FrameworkUtils.isSessionDataStorageOptimizationEnabled();
```

Comment:
do we really need this isSessionDataStorageOptimizationEnabled method in utils. Can't we follow the approach we used for JDBCPersistenceManager.SessionDataPersist.Temporary